### DISCUSSION TITLE
Grafana Panel Load Performance - XHR v Fetch

### DISCUSSION THREAD

**Discussion by joepallotta39** [grafana/grafana#74187] [Alerting]
Hi All, Looking for some guidance on improving panel load times after our Grafana 9 migration. Grafana render ajax/callback requests now use the fetch/promise request, causing a significant performance degradation on the overall panel loads (Dashboards > 6 panels). Browsers support 6 concurrent XHR requests, which creates a bottleneck with the fetch api given the promise waits for all the parallel requests to complete before returning a response.
It seems the fetch api is the standard approach for making web service calls so I assume I am missing something here. Any recommendations on possible solutions, or configuration/setup options I’m not aware of (besides forking Grafana)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424).

We do not currently have plans to switch our request architecture at this time. I am going to close this discussion, but feel free provide more concrete replication steps on the bottleneck you're experiencing and create an [issue](https://github.com/grafana/grafana/issues) for it, and our engineers can take a look. Thank you!
